consensus/spos/worker.go

```diff
@@ -73,6 +73,9 @@ type Worker struct {
 	publicKeySize       int
 	publicKeyBitmapSize int
 	cancelFunc          func()
+
+	mutPkConsensusMessages sync.RWMutex
```

Comment:
the way the code looks now, a better approach (IMHO) would have been to extract these 2 fields & the methods `Worker.addMessageTypeToPublicKey`, `Worker.isMessageTypeLimitReached` and `Worker.ResetConsensusMessages` into a separate component (in this way Worker struct wouldn't grow so much)